c++/test/dsets.cpp

```diff
@@ -1315,6 +1288,63 @@ test_virtual()
     }
 } // test_virtual
 
+/*-------------------------------------------------------------------------
+ * Function:    test_operator
+ *
+ * Purpose      Tests DataSet::operator=
+ *
+ * Return       Success: 0
+ *
+ *              Failure: -1
+ *-------------------------------------------------------------------------
+ */
+static herr_t
+test_operator(H5File &file)
+{
+    SUBTEST("DataSet::operator=");
+
+    try {
+        // Create a data space
+        hsize_t dims[2];
+        dims[0] = 256;
+        dims[1] = 512;
+        DataSpace space(2, dims, NULL);
+
+        // Create a dataset using the default dataset creation properties.
+        // We're not sure what they are, so we won't check.
+        DataSet dataset = file.createDataSet(DSET_OPERATOR, PredType::NATIVE_DOUBLE, space);
+
+        // Add a comment to the dataset
+        file.setComment(DSET_OPERATOR, "Dataset using operator=");
+
+        // Close the dataset
+        dataset.close();
+
+        // Re-open the dataset
+        DataSet another_dataset(file.openDataSet(DSET_OPERATOR));
+
+        // Try operator= to make another dataset
+        DataSet copied_dataset = another_dataset;
+
+        hid_t copied_dataset_id  = copied_dataset.getId();
+        hid_t another_dataset_id = another_dataset.getId();
```

Comment:
I built and ran the test, and these two lines caused these unused-variable warnings:
/home/hdftest/snapshots-hdf5_1_12_1/current/c++/test/dsets.cpp: In function ‘herr_t test_operator(H5::H5File&)’:
/home/hdftest/snapshots-hdf5_1_12_1/current/c++/test/dsets.cpp:1329:15: warning: unused variable ‘copied_dataset_id’ [-Wunused-variable]
         hid_t copied_dataset_id  = copied_dataset.getId();
               ^
/home/hdftest/snapshots-hdf5_1_12_1/current/c++/test/dsets.cpp:1330:15: warning: unused variable ‘another_dataset_id’ [-Wunused-variable]
         hid_t another_dataset_id = another_dataset.getId();

Can you use the ids for some trivial purpose?  maybe to close them, or to get the names?